app/src/main/java/io/gnosis/safe/ui/transaction/TransactionsViewModel.kt

```diff
@@ -1,103 +1,122 @@
 package io.gnosis.safe.ui.transaction
 
 import androidx.annotation.StringRes
-import io.gnosis.data.models.Page
+import androidx.lifecycle.viewModelScope
+import androidx.paging.PagingData
+import androidx.paging.cachedIn
+import androidx.paging.insertSeparators
 import io.gnosis.data.models.Safe
 import io.gnosis.data.models.Transaction
 import io.gnosis.data.models.TransactionStatus
 import io.gnosis.data.repositories.SafeRepository
-import io.gnosis.data.repositories.TransactionRepository
 import io.gnosis.safe.R
 import io.gnosis.safe.ui.base.AppDispatchers
 import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.transaction.list.TransactionPagingRepository
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.map
 import pm.gnosis.model.Solidity
 import javax.inject.Inject
 
 class TransactionsViewModel
 @Inject constructor(
-    private val transactionRepository: TransactionRepository,
+    private val transactionsPagingRepository: TransactionPagingRepository,
     private val safeRepository: SafeRepository,
     appDispatchers: AppDispatchers
 ) : BaseStateViewModel<TransactionsViewState>(appDispatchers) {
 
+    private var currentSafeAddress: Solidity.Address? = null
+    private var currentSafeTxItems: Flow<PagingData<TransactionView>>? = null
```

Comment:
Does this need to be a `var`? Couldn't we have a `flow` that starts empty and then emits as new pages of data become available?

Reply:
it is used for caching

Reply:
I understand that. But if you overwrite the `flow` every single time a new value is available, then you don't need a `flow` at all, I would say. Then you can just have a `PagingData<TransactionView>>?` var.

Reply:
To me it seems like it all boils down to deciding whether this is a cache or a stream?

Reply:
new PagingData's are emitted when we load new pages. We overwrite the flow only if safe address changes

Reply:
but does it matter that we have a new instance of `flow` each time? To me it seems similar to how we use `state`, in which case we can reuse the instance as it is more like a message bus rather than anything else.

Reply:
yes, because new instance of flow is created by a paging library